pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -1026,6 +1037,86 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
 	return &registriesIgn, nil
 }
 
+func (ctrl *Controller) createDefaultContainerRuntimeMC() error {
+	// Check if the crio-default-container-runtime config map exists in the openshift-machine-config-operator namespace
+	defaultContainerRuntimeCM, err := ctrl.kubeClient.CoreV1().ConfigMaps(ctrl.namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error checking for %s config map: %w", configMapName, err)
+	}
+	// If the crio-default-container-runtime config map exists, the MC was already created, so skip creating it again.
+	if defaultContainerRuntimeCM != nil && !errors.IsNotFound(err) {
+		return nil
+	}
+
+	sel, err := metav1.LabelSelectorAsSelector(metav1.AddLabelToSelector(&metav1.LabelSelector{}, builtInLabelKey, ""))
+	if err != nil {
+		return err
+	}
+	// Find all the MachineConfigPools
+	mcpPoolsAll, err := ctrl.mcpLister.List(sel)
+	if err != nil {
+		return err
+	}
+
+	// Create the crio-default-container-runtime MC for all the available pools
+	for _, pool := range mcpPoolsAll {
+		managedKey := getManagedKeyDefaultContainerRuntime(pool)
+		mc, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), managedKey, metav1.GetOptions{})
```

Comment:
should this be a ContainerRuntimeConfig instead of a MC?

Reply:
I think this can work fine. managedKey is the name of the mc we are trying to check if it has been created in the current pool. we don't directly create ContainerRuntimeConfig through the API server.

Reply:
@rbaturov may need it in the API server for the NTO to ensure this works, we're talking about it offline now. Also: if we want to avoid an additional reboot update, won't we need to use a ctrcfg to dedup?

Reply:
oh I see, you mean that can avoid the reboot by creating a /etc/crio/crio.conf.d/01-ctrcfg-defaultRuntime for dup configuration.

Reply:
yes, if the configurations in the same file stay the same we don't reboot.

Reply:
If the concern is only aout the extra reboot, instead of creating a ContainerRuntimeConfig, I think in this PR we can write our default configuration to the file `/etc/crio/crio.conf.d/01-ctrcfg-defaultRuntime`, so that a customer-created ContainerRuntimeConfig with the same configuration can avoid draining the node.



> @rbaturov may need it in the API server for the NTO to ensure this works

@haircommander  I am not sure about their usecase, we must ship a containerruntimeconfig CR so they can get it from the API server?

Reply:
yeah the NTO needs to see what the default runtime is to use the same one for the performance profile runtime class

Reply:
I understand their use case is that we just need to ship a default `ContainerRuntimeConfig` CR; we don't have to create the CR or MC in MCO in this PR.

I think it should be similar to the default `ClusterImagePolicy`
we add the manifest of the CR, and MCO can process it by `syncContainerRuntimeConfig` and `syncContainerRuntimeConfigBootstrap`.  Is this the correct approach?

I'm not sure where we should add that manifest.

Reply:
something I'm thinking about is that we can't create the ctrcfg for all branches, we can only do so in 4.17 so we should do it in this PR. otherwise, we'll just update the default runtime in 4.18 when we upgrade, which goes against the aims of this. does that make sense?

Reply:
Yes, I agree that we only need the ContainerRuntimeConfig in 4.17.  I'm just not sure if MCO is the best place to create it, as doing so might cause the sync handler to make a recursive call and process the object on its own. But it would work.

Reply:
To add to this, creating a containerruntime config object directly is not really an option, since this will conflict with any user-provided ones. CRC objects do not merge like MCs.